pkg/controller/networkpolicy/networkpolicy_controller_perf_test.go

```diff
@@ -328,30 +358,40 @@ func getRandomNodeName() string {
 }
 
 // getXObjects calls the provided getObjectsFunc x times and aggregate the objects.
-func getXObjects(x int, getObjectsFunc func() ([]*corev1.Namespace, []*networkingv1.NetworkPolicy, []*corev1.Pod)) ([]*corev1.Namespace, []*networkingv1.NetworkPolicy, []*corev1.Pod) {
+func getXObjects(x int, getObjectsFunc func() ([]*corev1.Namespace, []*networkingv1.NetworkPolicy, []*v1alpha1.NetworkPolicy, []*corev1.Pod, []*v1alpha2.ExternalEntity)) ([]*corev1.Namespace, []*networkingv1.NetworkPolicy, []*v1alpha1.NetworkPolicy, []*corev1.Pod, []*v1alpha2.ExternalEntity) {
 	var namespaces []*corev1.Namespace
 	var networkPolicies []*networkingv1.NetworkPolicy
+	var anp []*v1alpha1.NetworkPolicy
 	var pods []*corev1.Pod
+	var externalEntities []*v1alpha2.ExternalEntity
 	for i := 0; i < x; i++ {
-		newNamespaces, newNetworkPolicies, newPods := getObjectsFunc()
+		newNamespaces, newNetworkPolicies, newANP, newPods, newExternalEntities := getObjectsFunc()
 		namespaces = append(namespaces, newNamespaces...)
 		networkPolicies = append(networkPolicies, newNetworkPolicies...)
+		anp = append(anp, newANP...)
 		pods = append(pods, newPods...)
+		externalEntities = append(externalEntities, newExternalEntities...)
 	}
-	return namespaces, networkPolicies, pods
+	return namespaces, networkPolicies, anp, pods, externalEntities
 }
 
-func toRunTimeObjects(namespaces []*corev1.Namespace, networkPolicies []*networkingv1.NetworkPolicy, pods []*corev1.Pod) []runtime.Object {
-	objs := make([]runtime.Object, 0, len(namespaces)+len(networkPolicies)+len(pods))
+func toRunTimeObjects(namespaces []*corev1.Namespace, networkPolicies []*networkingv1.NetworkPolicy, anp []*v1alpha1.NetworkPolicy, pods []*corev1.Pod, externalEntities []*v1alpha2.ExternalEntity) []runtime.Object {
```

Comment:
After unifying the policy and entities parameters, you can use type assertion to classify k8sObjects and crdObjects:
```
for _, obj := range networkPolicies {
	switch policy := obj.(type) {
	case *networkingv1.NetworkPolicy:
		k8sObjs = append(k8sObjs, policy)
	case *crdv1alpha1.NetworkPolicy:
		crdObjs = append(crdObjs, policy)
	}
}
```

Reply:
done, thanks.